### PR TITLE
[cluster-test] Add packet loss experiment and effect

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

Adds `PacketLoss`, `RemoveNetworkEffect` effects and an experiment for injecting a given amount of packet loss across a given number of instances

## Test Plan

Ran this experiment on my test cluster.

```
[ec2-user:validator@ip-10-0-0-212 ~]$ ~/libra/target/cluster_test_docker_builder/cluster-test --workplace=kush --packet-loss-experiment --packet-loss-percent-instances=50 --packet-loss-percent=20
Oct 28 22:33:38.029 INFO Discovered 4 peers
Oct 28 22:33:39.248 INFO Log tail thread started in 1219 ms
PacketLoss 20.00% for 57ff8374(10.0.3.130)
Oct 28 22:33:39.315 INFO Starting experiment Packet Loss 20.00% [57ff8374(10.0.3.130), 1e5d5a74(10.0.6.55), ]
PacketLoss 20.00% for 1e5d5a74(10.0.6.55)
RemoveNetworkEffects for 57ff8374(10.0.3.130)
RemoveNetworkEffects for 1e5d5a74(10.0.6.55)
Oct 28 22:35:04.343 INFO Experiment finished, waiting until all affected validators recover
Oct 28 22:35:09.348 INFO Experiment completed
```

image

Related to #1198

### CODE REVIEW

**Comment by andll** [comment]
Thanks!

I will look in details, but so far what I noticed:
Can you use debug/info macro instead of println? `info` to always print it (default log level) and `debug` if you don't want to see it by default (but can see it if run with RUST_LOG=debug)

File: testsuite/cluster-test/src/experiments/packet_loss_random_validators.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            panic!(
+                "count {} should be <= number of instances {}",
+                count,
+                cluster.instances().len()
+            );
+        }
+        let mut instances = Vec::with_capacity(count);
+        let mut all_instances = cluster.instances().clone();
+        let mut rnd = rand::thread_rng();
+        for _i in 0..count {
```
Comment (andll) [inline testsuite/cluster-test/src/experiments/packet_loss_random_validators.rs]: I added `Cluster::split_n_random` in #1531, do you mind switching to it at some point?

https://github.com/libra/libra/pull/1531/files#diff-3b9be419cfe902ff91ce042d9fb56865R145

**Comment by andll** [comment]
I also think we will need to add `cleanup()` method and `--cleanup` command to ClusterTestRunner that would clean any pending effects on cluster in case some experiment crashed.
We could then call `cleanup()` before methods like `run()` and `run-once()`

I am ok if we do those to things in follow up PRs, leaving it up to you

 @bors-libra delegate+

**Comment by andll** [comment]
One thing though - you PR seem to have 2 commits, can you squash them before merging?

**Comment by ankushagarwal** [comment]
I have used `git commit --fixup` for the new commit. bors-libra will automatically squash that into a single commit before merging.

I do the followups in a separate PR.

@bors-libra r+

**Pull request closed without merging**